libraries/appbase/CMakeLists.txt

```diff
@@ -21,11 +21,11 @@ set( Boost_USE_STATIC_LIBS ON CACHE STRING "ON or OFF" )
 if( APPLE )
   # Apple Specific Options Here
   message( STATUS "Configuring AppBase on OS X" )
-  set( CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -std=c++14 -stdlib=libc++ -Wall -Wno-conversion -Wno-deprecated-declarations" )
+  set( CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -std=c++17 -stdlib=libc++ -Wall -Wno-conversion -Wno-deprecated-declarations" )
```

Comment:
Can you please change to "more modern" way https://cmake.org/cmake/help/v3.13/prop_tgt/CXX_STANDARD.html
So, something like this 
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)